runelite-client/src/main/java/net/runelite/client/plugins/nightmarezone/NightmareZonePlugin.java

```diff
@@ -170,8 +216,51 @@ private void checkAbsorption()
 		}
 	}
 
+	private void calculateHitpointPercentage()
```

Comment:
So this only assumes that you are using the rapid heal. This is probably good enough for most people as you want to have the regen be spread out as much as possible. I am not sure if people use the regen bracelet/hp cape to reset the counter or if they would want it, but otherwise this seems to work.

Reply:
Yes the program doesn't take into consideration the hp cape and regen bracelet because as far as I know no one uses it in Nightmare zone (this code is a replica of the one used in the plugin that shows the current process of hp regen on the orb).

Reply:
Like I said its most likely good enough. Just thought I would make sure the other functionality is known.

Reply:
It should work with hp cape/regen brace because they heal double, not halve the restore time.